flycheck.el

```diff
@@ -4288,6 +4304,10 @@ symbols in the command."
           ;; example for such a conflict.
           (setq process (apply 'start-process (format "flycheck-%s" checker)
                                nil program args))
+          (when (and (flycheck-checker-standard-input-p checker)
+                     (process-live-p process))
+            (process-send-region process (point-min) (point-max))
```

Comment:
Please use `0` and `(1+ (buffer-size))` instead of `point-min` and `point-max` respectively, to make sure that Flycheck always sends the entire buffer contents to the subprocess, even in narrowed buffers.